src/Controls/src/Core/Handlers/Items/iOS/TemplatedCell.cs

```diff
@@ -113,6 +113,15 @@ protected void Layout(CGSize constraints)
 			_size = rectangle.Size;
 		}
 
+		public override void PrepareForReuse()
+		{
+			if (PlatformHandler?.VirtualView is View view)
+			{
```

Comment:
This should also unhook the `MeasureInvalidated` handler, otherwise all of the binding updates that happen when the context is updated will invoke the handler. 

Also, you can simplify the check on line 175 if you're doing this; BindingContext will always be null there.